### ISSUE TITLE
Pie Plot not showing up

### ISSUE THREAD

**Issue by JulesBlubb** [GenieFramework/Stipple.jl#33]
Hi, 
I want to do a simple Pie-chart. The labels are displayed, but not the pie-chart. What am I doing wrong?
```julia
using Genie, Genie.Renderer.Html, Genie.Router

using Stipple
using StippleUI, StippleUI.Heading, StippleUI.Dashboard
using StippleCharts, StippleCharts.Charts

using DataFrames

data = DataFrame(Costs = [44, 55, 13, 43, 22])

Base.@kwdef mutable struct Dashboard <: ReactiveModel
    plot_options::PlotOptions = PlotOptions(chart_type=:pie, chart_width=380,labels= ["Team A", "Team B", "Team C", "Team D", "Team E"])
    piechart::R{Vector{PlotSeries}} = []
end

# UI
Stipple.register_components(Dashboard, StippleCharts.COMPONENTS)
const model = Stipple.init(Dashboard())

on(model.piechart) do _
    model.piechart[] = [PlotSeries("name", PlotData(data.Costs))]
end

function ui()
    dashboard(vm(model), [
        h3("Example Pie Plot")
        row(cell(class="st-module", plot(@data(:piechart); options=:plot_options)))
    ])
end

route("/") do
    ui() |> html
end

up(rand((8000:9000)), open_browser=true)
```

Thanks!! :)

**Comment by essenciary** [comment]
@JulesBlubb What is the output of `julia> versioninfo()` and `pkg> st` please?

**Comment by JulesBlubb** [comment]
```
Julia Version 1.3.1
Commit 2d5741174c (2019-12-30 21:36 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)
```
```
  [4acbeb90] Stipple v0.7.1
  [30ddb3f0] StippleCharts v0.4.0
  [a3c5d34a] StippleUI v0.2.3
  [c43c736e] Genie v1.10.0
  [a93c6f00] DataFrames v0.21.8

```

**Comment by essenciary** [comment]
You're a few versions behind on all the packages. Can you please update to latest version and see how it goes.

**Comment by JulesBlubb** [comment]
I have updated all packages, but stayed with Julia 1.3.1. Do I need Julia 1.5?
Now the labels are also gone. Only the heading is shown.

**Comment by essenciary** [comment]
Thanks - can you please check the JS console in the browser's developer tools and see if there are any errors?

**Comment by JulesBlubb** [comment]
![Bildschirmfoto vom 2021-03-05 23-18-48](https://user-images.githubusercontent.com/5750890/110179978-45747e80-7e09-11eb-9166-4acd79aa2b0e.png)

**Comment by essenciary** [comment]
Thanks - yes, I was able to reproduce the problem, a regression was introduced in v0.8. I'll release a patch soon.

**Comment by essenciary** [comment]
@JulesBlubb Sorry about that - I just tagged v0.9.4 which address the problem. Please make sure you're running that version (should be registered in a few minutes). Also, beware that there's yet another regression introduced in v0.8 which is not critical and we're working on debugging it, per #34

**Comment by JulesBlubb** [comment]
Thank you :) 
The German Credit demo works again. My code from above works now for type=:bar but not for pie charts. Again the legend is showing up, but not the plot. Am I missing something?

**Comment by essenciary** [comment]
I've reproduced your results - now it seems that a different issue is affecting only the pie charts. I'm looking into it - hopefully it's the last :)

**Comment by hhaensel** [comment]
The problem is the data type of `model.piechart`. It expects a simple vector
```
using Stipple, StippleUI, StippleCharts

using DataFrames

data = DataFrame(Costs = [44, 55, 13, 43, 22])

OptDict = Dict{Symbol, Any}
opts(;kwargs...) = OptDict(kwargs...)

Base.@kwdef mutable struct Dashboard <: ReactiveModel
    plot_options::PlotOptions = PlotOptions(chart_type=:pie, chart_width=380,labels= ["Team A", "Team B", "Team C", "Team D", "Team E"])
    # alternative definition
    # plot_options::R{OptDict} = opts(
    #     chart = opts(type = :pie, width = 380),
    #     labels = ["Team A", "Team B", "Team C", "Team D", "Team E"]
    # )
    piechart::R{Vector{Float64}} = Float64[]
    newplot::R{Bool} = false
end

# UI
Stipple.register_components(Dashboard, StippleCharts.COMPONENTS)

model = Stipple.init(Dashboard())

onbutton(model.newplot) do
    @info "new plot"
    model.piechart[] = data.Costs .+ rand(-20:20, length(nrow(data)))
end

function ui()
    dashboard(vm(model), [
        h3("Example Pie Plot")
        row(cell(class="st-module", plot(:piechart; options = :plot_options)))
        row(cell(btn("newPlot", @click(:newplot), color="primary")))
    ])
end

route("/") do
    ui() |> html
end

up()
```

**Comment by hhaensel** [comment]
Note the new `onbutton()` which will only work with current master. Otherwise just do
```
on(model.newplot) do pressed
    pressed || return
    @info "new plot"
    model.piechart[] = data.Costs .+ rand(-20:20, length(nrow(data)))
    model.newplot[] = false
end
```
@essenciary Shall I register a new version?

**Comment by essenciary** [comment]
Yes, that's one of the things, pie works with a simple vector, and this needs to be documented. But some other things are still not working and I'm not sure why. Still working on it. I've added a few pie charts features yesterday, in the latest release, but things like donut or size settings don't seem to work. 

My test code is 

```julia
using Genie, Genie.Renderer.Html, Genie.Router

using Stipple
using StippleUI
using StippleCharts

Base.@kwdef mutable struct Dashboard <: ReactiveModel
    plot_options::PlotOptions = PlotOptions(chart_type=:pie, chart_width=380, 
                                            stroke_show = false, plot_options_pie_size = 380, 
                                            labels=["Team A", "Team B", "Team C", "Team D", "Team E"])
    piechart::R{Vector{Int}} = [44, 55, 13, 43, 22]
end

# UI
Stipple.register_components(Dashboard, StippleCharts.COMPONENTS)

model = Stipple.init(Dashboard())

function ui()
    page(vm(model), [
        h3("Example Pie Plot")

        row(
          cell(class="st-module", [
            plot(@data(:piechart), options=:plot_options)
          ])
        )
    ])
end

route("/") do
    ui() |> html
end

up(rand((8000:9000)), open_browser=true)
```

@hhaensel Yes please, go for it!

**Comment by hhaensel** [comment]
Your example works for me ...
![image](https://user-images.githubusercontent.com/31985040/110493838-4819f100-80f3-11eb-99d5-50f975b5acde.png)

**Comment by hhaensel** [comment]
@essenciary 
What do you think about naming the current master 0.9.x-DEV?
I started with that convention, because I think it is good to distinguish master and latest release in the `project.toml`

**Comment by essenciary** [comment]
@hhaensel So keep `DEV` while we work on the version and drop it when we register/tag? 

It works? And it renders a donut by default? What is the output of `pkg> st` for you please?

**Comment by hhaensel** [comment]
re-evaluated with a completely empty project, Julia 1.6.0-rc1
```
(Temp) pkg> st
      Status `C:\Temp\Project.toml`
  [4acbeb90] Stipple v0.9.6
  [30ddb3f0] StippleCharts v0.5.0
  [a3c5d34a] StippleUI v0.2.4
```
works!

**Comment by hhaensel** [comment]
> @hhaensel So keep `DEV` while we work on the version and drop it when we register/tag?

Exactly, and set the next version and append DEV for the following work

There is a nice tool in VS Code that does all that automatically...

**Comment by essenciary** [comment]
Any idea as what might've broken our releases/tagging? Eg Stipple, the latest release (which was done automatically bu Julia Registrator) is v0.9.3. The next version was v0.9.4-DEV and that's when releases/tags stopped working.

**Comment by essenciary** [comment]
Same for StippleCharts, auto tagging / releasing stopped with v0.4.2-DEV

**Comment by JulesBlubb** [comment]
Your code works for me and I receive beautiful pie charts. Thanks a lot! :)

**Comment by hhaensel** [comment]
I've created v0.5.1-DEV and released v0.5.0 of StippleCharts. So all projects follow the new version convention.
I hope that with the next release we will have auto-tagging againg in place.
